bika/lims/browser/auditlog.py

```diff
@@ -0,0 +1,309 @@
+# -*- coding: utf-8 -*-
+
+import collections
+import json
+
+from bika.lims import api
+from bika.lims import bikaMessageFactory as _
+from bika.lims import logger
+from bika.lims.browser.bika_listing import BikaListingView
+from bika.lims.interfaces import IAuditable
+from bika.lims.utils import t
+from plone.memoize import view
+from Products.CMFPlone.i18nl10n import ulocalized_time
+from Products.Five.browser import BrowserView
+from Products.Five.browser.pagetemplatefile import ViewPageTemplateFile
+
+
+class HasAuditLog(BrowserView):
+    """Helper View to check if the context provides the IAuditable interface
+    """
+    def __call__(self):
+        return IAuditable.providedBy(self.context)
+
+
+class AuditLogView(BikaListingView):
+    """Audit View
+    """
+    template = ViewPageTemplateFile("templates/audit.pt")
+    diff_template = ViewPageTemplateFile("templates/audit_diff.pt")
+
+    def __init__(self, context, request):
+        super(AuditLogView, self).__init__(context, request)
+
+        self.show_select_column = False
+        self.show_search = False
+        self.pagesize = 5
+
+        self.icon = "{}/{}/{}".format(
+            self.portal_url,
+            "++resource++bika.lims.images",
+            "%s_big.png" % context.portal_type.lower())
+
+        self.title = "{} Log".format(api.get_title(context))
+
+        self.columns = collections.OrderedDict((
+            ("version", {
+                "title": _("Version"), "sortable": False}),
+            ("time", {
+                "title": _("Time"), "sortable": False}),
+            ("modified", {
+                "title": _("Date Modified"), "sortable": False}),
+            ("actor", {
+                "title": _("Actor"), "sortable": False}),
```

Comment:
We also need to display the full name of the actor when the modification took place.